share/revng/test/tests/analysis/DetectABI/x86_64/propagate.S.yml.filecheck

```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) rev.ng Labs Srl. See LICENSE.md for details.
+#
+
+CHECK:      Functions:
+CHECK:        - Entry: "0x400660:Code_x86_64"
+CHECK-NEXT:     Prototype: "/Types/CABIFunctionType-5831792892140344958"
```

Comment:
We can't hard code addresses and type IDs.

Consider using `revng model compare`, see the other usages, if they make sense to you or ping me for some support.
`revng model compare` basically enables you verify the "shape" of the model (e.g., two functions sharing the same prototype) without relying on hardcoding addresses or type IDs. Basically your ground truth should look something like this:

```
Functions:
- CustomName: caller
  Prototype: "/Types/CABIFunctionType-66666666666"
DynamicFunctions:
- OriginalName: puts
  Prototype: "/Types/CABIFunctionType-66666666666"
Types:
- Type: CABIFunctionType
  ID: 66666666666
```

`revng model compare` is smart enough to detect if the model you feed to it is compatible with the one above, even if the type ID is not 66666666666 and even if the model above is invalid (functions are missing the mandatory `Entry` attribute).

Reply:
You explicitly asked for FileCheck tests, so that's why I decided to hardcode type id seed and compare specific values.

Reply:
We can probably make it work with `FileCheck` too, if you put a placeholder in place of the ID, but then the test case is not super robust w.r.t. to the order of the functions.

It's better to switch to `revng model compare` which enables you to write more "semantically rich" ground truth.